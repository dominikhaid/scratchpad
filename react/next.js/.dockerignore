node_modules
npm-debug.log
/.git
/docker-backups
/node_modules
/docker
yarn.lock
