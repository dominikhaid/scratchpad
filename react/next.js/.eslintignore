node_modules
.vscode
gitignore
log
uploads
public/images
public/icons