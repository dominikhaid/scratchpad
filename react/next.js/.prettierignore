node_modules
log
gitignore
uploads