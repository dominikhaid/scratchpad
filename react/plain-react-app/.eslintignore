node_modules
.vscode
gitignore
log
uploads
public/images
public/icons
**/*.spec.js
cypress/plugins
cypress/support
**/reportWebVitals.ts
**/tailwind/*.*js
**/*.config.js
**/reportWebVitals.js
