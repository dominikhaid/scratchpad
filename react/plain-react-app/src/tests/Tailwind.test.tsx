import {render, screen, cleanup} from '@testing-library/react';
import Tailwind from '../pages/Tailwind';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('should take a snapshot', () => {
  const {asFragment} = render(<Tailwind />);
  expect(asFragment(<Tailwind />)).toMatchSnapshot();
});

test('renders learn react link', () => {
  render(<Tailwind />);
  const linkElement = screen.getByText(/Tailwind Stylesheet/i);
  expect(linkElement).toBeInTheDocument();
});
