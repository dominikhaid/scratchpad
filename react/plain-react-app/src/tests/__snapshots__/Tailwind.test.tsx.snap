// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed w-full uppercase spaced p-2xl bg-gray-darkest"
  >
    <a
      class="text-white "
      href="#colors"
    >
      colors
    </a>
    <a
      class="text-white"
      href="#text"
    >
      text
    </a>
    <a
      class="text-white"
      href="#padding"
    >
      padding
    </a>
    <a
      class="text-white"
      href="#margin"
    >
      margin
    </a>
    <a
      class="text-white"
      href="#elements"
    >
      elements
    </a>
    <a
      class="text-white"
      href="#shadows"
    >
      shadows
    </a>
    <a
      class="text-white"
      href="#colors"
    >
      colors
    </a>
    <a
      class="text-white"
      href="#containers"
    >
      Container
    </a>
  </div>
  <h1>
    Tailwind Stylesheet
  </h1>
  <section
    class="items-center grid-cols-4 md:grid lg:grid xl:grid bg-gray-lightest gap-lg p-4xl sm:grid-flow-col-dense"
    id="colors"
  >
    <h1
      class="col-span-4 text-center uppercase p-md mt-6xl"
    >
      Colors
    </h1>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-primary-lightest"
      />
      <div
        class="flex-auto bg-primary-light"
      />
      <div
        class="flex-auto bg-primary"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          Primary
        </h3>
      </div>
      <div
        class="flex-auto bg-primary-dark"
      />
      <div
        class="flex-auto bg-primary-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-secondary-lightest"
      />
      <div
        class="flex-auto bg-secondary-light"
      />
      <div
        class="flex-auto bg-secondary"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          Secondary
        </h3>
      </div>
      <div
        class="flex-auto bg-secondary-dark"
      />
      <div
        class="flex-auto bg-secondary-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-gray-lightest"
      />
      <div
        class="flex-auto bg-gray-light"
      />
      <div
        class="flex-auto bg-gray"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          Gray
        </h3>
      </div>
      <div
        class="flex-auto bg-gray-dark"
      />
      <div
        class="flex-auto bg-gray-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-green-lightest"
      />
      <div
        class="flex-auto bg-green-light"
      />
      <div
        class="flex-auto bg-green"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          Green
        </h3>
      </div>
      <div
        class="flex-auto bg-green-dark"
      />
      <div
        class="flex-auto bg-green-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-red-lightest"
      />
      <div
        class="flex-auto bg-red-light"
      />
      <div
        class="flex-auto bg-red"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          Red
        </h3>
      </div>
      <div
        class="flex-auto bg-red-dark"
      />
      <div
        class="flex-auto bg-red-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-blue-lightest"
      />
      <div
        class="flex-auto bg-blue-light"
      />
      <div
        class="flex-auto bg-blue"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          blue
        </h3>
      </div>
      <div
        class="flex-auto bg-blue-dark"
      />
      <div
        class="flex-auto bg-blue-darkest"
      />
    </div>
    <div
      class="flex border-2 shadow-xl bg-gray-light p-lg m-xs"
    >
      <div
        class="flex-auto bg-yellow-lightest"
      />
      <div
        class="flex-auto bg-yellow-light"
      />
      <div
        class="flex-auto bg-yellow"
      >
        <h3
          class="text-center text-white uppercase p-md"
        >
          yellow
        </h3>
      </div>
      <div
        class="flex-auto bg-yellow-dark"
      />
      <div
        class="flex-auto bg-yellow-darkest"
      />
    </div>
  </section>
  <section
    class="flex flex-wrap items-center justify-center bg-secondary-light p-4xl"
    id="text"
  >
    <h1
      class="flex-none col-span-4 text-center uppercase p-md mt-6xl w-100"
      data-testid="text-head"
    >
      Text
    </h1>
    <article
      class="prose prose-lg p-lg"
    >
      <h1>
        Garlic bread with cheese: What the science tells us
      </h1>
      <p>
        For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf htmlFor Halloween.
      </p>
      <p>
        But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
      </p>
      <h2>
        Garlic bread with cheese: What the science tells us
      </h2>
      <p>
        For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf htmlFor Halloween.
      </p>
    </article>
    <article
      class="prose prose-lg p-lg"
    >
      <h3>
        Garlic bread with cheese: What the science tells us
      </h3>
      <p>
        But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
      </p>
      <h4>
        Garlic bread with cheese: What the science tells us
      </h4>
      <p>
        But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
      </p>
      <h5>
        Garlic bread with cheese: What the science tells us
      </h5>
      <p>
        But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
      </p>
    </article>
  </section>
  <section
    class="flex flex-wrap items-center justify-center bg-primary-light p-4xl"
    id="padding"
  >
    <h1
      class="text-center uppercase p-md mt-6xl min-w-100"
    >
      Padding
    </h1>
    <div
      class="self-center border-2 shadow-xl bg-primary p-xs m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-xs
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-sm m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-sm
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-md m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-md
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-primary p-lg m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-lg
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-primary p-xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-primary p-2xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-2xl
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-3xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-3xl
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-4xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-4xl
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-5xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-5xl
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-6xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-6xl
        </h3>
      </div>
    </div>
    <div
      class="self-center border-2 shadow-xl bg-primary p-7xl m-xs"
    >
      <div
        class="min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          p-7xl
        </h3>
      </div>
    </div>
  </section>
  <section
    class="flex flex-wrap items-center justify-center bg-secondary-light p-4xl"
    id="margin"
  >
    <h1
      class="text-center uppercase p-md mt-6xl min-w-100"
    >
      Margin
    </h1>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-xs bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-xs
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-sm bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-sm
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-md bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-md
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-lg bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-lg
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-xl min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-2xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-2xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-3xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-3xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-4xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-4xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-5xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-5xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-6xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-7xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          m-7xl
        </h3>
      </div>
    </div>
  </section>
  <section
    class="items-center grid-cols-2 bg-primary-light md:grid lg:grid xl:grid gap-lg p-4xl"
    id="elements"
  >
    <h1
      class="col-span-2 text-center uppercase p-md mt-6xl min-w-100"
    >
      Elements
    </h1>
    <section
      class="grid items-center grid-cols-2 gap-lg p-4xl"
    >
      <h2
        class="text-center uppercase p-md"
      >
        Form
      </h2>
      <div />
      <form>
        <fieldset
          class="labeled"
        >
          <label
            for="intext"
          >
            Text:
          </label>
          <input
            id="intext"
            type="text"
            value="Text"
          />
        </fieldset>
        <fieldset
          class="labeled"
        >
          <label
            for="innr"
          >
            Nr:
          </label>
          <input
            id="innr"
            placeholder="123"
            type="number"
          />
        </fieldset>
        <fieldset
          class="labeled"
        >
          <label
            for="inemail"
          >
            Email:
          </label>
          <input
            id="inemail"
            placeholder="example@test.com"
            type="email"
          />
        </fieldset>
        <fieldset
          class="labeled"
        >
          <label
            for="inpw"
          >
            Password:
          </label>
          <input
            id="inpw"
            type="password"
            value="1234"
          />
        </fieldset>
        <fieldset>
          <label
            for="jetzt"
          >
            Val1
          </label>
          <input
            id="jetzt"
            name="radiogrp"
            type="radio"
            value="current"
          />
          <label
            for="heute"
          >
            Val2
          </label>
          <input
            id="heute"
            name="radiogrp"
            type="radio"
            value="tomorrow"
          />
          <label
            for="fiveday"
          >
            Val3
          </label>
          <input
            id="fiveday"
            name="radiogrp"
            type="radio"
            value="fiveday"
          />
        </fieldset>
      </form>
    </section>
    <section>
      <h2
        class="text-center uppercase p-md"
      >
        Spaced
      </h2>
      <div
        class="spaced min-h-75 m-sm"
      >
        <div
          class="text-white bg-secondary-dark w-50"
        >
          <p
            class="text-center p-xl"
          >
            spaced
          </p>
        </div>
        <div
          class="text-white bg-secondary-darkest w-50"
        >
          <p
            class="text-center p-xl"
          >
            spaced
          </p>
        </div>
      </div>
      <div
        class="spaced-lg min-h-75 m-sm"
      >
        <div
          class="text-white bg-secondary-dark w-50"
        >
          <p
            class="text-center p-xl"
          >
            spaced-lg
          </p>
        </div>
        <div
          class="text-white bg-secondary-darkest w-50"
        >
          <p
            class="text-center p-xl"
          >
            spaced-lg
          </p>
        </div>
      </div>
    </section>
    <section
      class="modal demo"
    >
      <h2
        class="text-center uppercase p-md"
      >
        Modal
      </h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Pariatur atque saepe repudiandae magnam sequi laboriosam sint explicabo voluptas autem ipsa!
      </p>
    </section>
    <section
      class="p-4xl"
    >
      <h2
        class="text-center uppercase p-md"
      >
        Buttons
      </h2>
      <div
        class="items-center justify-center spaced"
      >
        <button
          class="btn-primray"
          data-testid="btn-primary"
        >
          Button
        </button>
        <button
          class="btn-secondary"
        >
          Button
        </button>
        <button
          class="btn-success"
        >
          Button
        </button>
        <button
          class="btn-primary"
          data-testid="btn-disabled"
          disabled=""
        >
          Button
        </button>
      </div>
    </section>
  </section>
  <section
    class="flex flex-wrap items-center justify-center bg-secondary-light min-h-75 p-4xl"
    id="shadow"
  >
    <h1
      class="text-center uppercase p-md mt-6xl min-w-100"
    >
      Shadows
    </h1>
    <div
      class="border-2 shadow-sm bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-sm
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-md bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-md
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-lg bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-lg
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-xl bg-secondary m-xs"
    >
      <div
        class="m-6xl min-h-100 bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-xl
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-inner bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-inner
        </h3>
      </div>
    </div>
    <div
      class="border-2 shadow-none bg-secondary m-xs"
    >
      <div
        class="m-6xl bg-gray-lightest"
      >
        <h3
          class="text-center p-md"
        >
          shadow-none
        </h3>
      </div>
    </div>
  </section>
  <section
    class="items-center bg-gray-lightest min-h-75"
    id="containers"
  >
    <h1
      class="text-center uppercase p-md mt-6xl min-w-100"
    >
      Container
    </h1>
    <div
      class="container-normal px-xs mt-lg mb-lb"
    >
      <div
        class="flex items-center justify-center bg-primary min-h-50"
      >
        <article
          class="prose prose-lg p-lg"
        >
          <h1>
            Normal px-xs
          </h1>
          <p>
            For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf htmlFor Halloween.
          </p>
        </article>
      </div>
    </div>
    <div
      class="container-normal px-4xl mt-lg mb-lb"
    >
      <div
        class="flex items-center justify-center bg-primary min-h-50"
      >
        <article
          class="prose prose-lg p-lg"
        >
          <h1>
            Normal px-4xl
          </h1>
          <p>
            For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf htmlFor Halloween.
          </p>
        </article>
      </div>
    </div>
    <div
      class="container-full mt-lg mb-lb"
    >
      <div
        class="flex items-center justify-center bg-primary min-h-50"
      >
        <article
          class="prose prose-lg p-lg"
        >
          <h1>
            Full/Fluid
          </h1>
          <p>
            For years parents have espoused the health benefits of eating garlic bread with cheese to their children, with the food earning such an iconic status in our culture that kids will often dress up as warm, cheesy loaf htmlFor Halloween.
          </p>
          <p>
            But a recent study shows that the celebrated appetizer may be linked to a series of rabies cases springing up around the country.
          </p>
        </article>
      </div>
    </div>
  </section>
  <footer />
</DocumentFragment>
`;
